plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TechInsightJsonRuleCheck } from '../types';
+import {
+  BooleanCheckResult,
+  FactChecker,
+  FactSchema,
+  TechInsightsStore,
+} from '@backstage/plugin-tech-insights-common';
+import { Engine } from 'json-rules-engine';
+import { TechInsightCheckRegistry } from './CheckRegistry';
+
+class JsonRulesEngineFactChecker
+  implements FactChecker<TechInsightJsonRuleCheck, BooleanCheckResult>
+{
+  private readonly checkRegistry: TechInsightCheckRegistry<TechInsightJsonRuleCheck>;
+  private repository: TechInsightsStore;
+  private readonly schemas: FactSchema[];
+
+  constructor(
+    schemas: FactSchema[],
+    checks: TechInsightJsonRuleCheck[],
+    repository: TechInsightsStore,
+  ) {
+    this.repository = repository;
+    this.schemas = schemas;
+    checks.forEach(check => this.validate(check));
+    this.checkRegistry = new TechInsightCheckRegistry(checks);
+  }
+
+  async check(entity: string, checkName: string): Promise<BooleanCheckResult> {
+    const engine = new Engine();
+    const techInsightCheck = this.checkRegistry.get(checkName);
+    const facts = await this.repository.getLatestFactsForRefs(
+      techInsightCheck.factRefs,
+      entity,
+    );
+
+    engine.addRule(techInsightCheck.rule);
```

Comment:
Maybe we can do some error handling here just in case we have undefined or null value for techInsight